libpod/networking_common.go

```diff
@@ -632,14 +640,22 @@ func (r *Runtime) ConnectContainerToNetwork(nameOrID, netName string, netOpts ty
 	return ctr.NetworkConnect(nameOrID, netName, netOpts)
 }
 
-// normalizeNetworkName takes a network name, a partial or a full network ID and returns the network name.
+// normalizeNetworkName takes a network name, a partial or a full network ID and
+// returns: 1) the network name and 2) the network_interface name for macvlan
+// and ipvlan drivers if the naming pattern is "device". Else, "".
 // If the network is not found an error is returned.
-func (r *Runtime) normalizeNetworkName(nameOrID string) (string, error) {
+func (r *Runtime) normalizeNetworkName(nameOrID, namingPattern string) (string, string, error) {
 	net, err := r.network.NetworkInspect(nameOrID)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
-	return net.Name, nil
+
+	netIface := ""
+	if namingPattern == "device" && (net.Driver == "macvlan" || net.Driver == "ipvlan") {
```

Comment:
Please add a comment on where "device" comes from, this should properly be a constant somewhere.

Also for macvlan and ipvlan it should use `types.MacVLANNetworkDriver` and `types.IPVLANNetworkDriver`

Reply:
Will update. The "device" string is used just in this function once, nowhere else. Do we still want it defined as a constant?

Reply:
Resolving this comment for now. If you think defining a constant is required, then let me know a good place for that.